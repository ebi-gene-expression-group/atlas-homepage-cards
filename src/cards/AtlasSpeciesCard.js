import React from 'react'
import EbiSpeciesIcon from 'react-ebi-species'
import styled from 'styled-components'
import cardPropTypes from './cardPropTypes'

const ClearLink = styled.a`
  border-bottom: none;
  
  :hover {
    border-bottom: none;
  }
`

const SpeciesCard = ({iconSrc, description, content}) =>
  <div style={{marginBottom:0, paddingBottom: `2rem`, textAlign: `center`}}>
    {
      description && description.url ?
        <a style={{fontSize: `6rem`, borderBottom: 0}} href={description.url}>
          <EbiSpeciesIcon species={iconSrc}/>
        </a> :
        <span style={{fontSize: `6rem`}}>
          <EbiSpeciesIcon species={iconSrc}/>
        </span>
    }

    {
      description &&
      <h5>
      {
        description.url ?
          <a href={description.url}>{description.text}</a> :
          description.text
      }
      </h5>
    }

    {
      Array.isArray(content) &&
      <p className={`experiments`}>{content[0].text} <br/>

        <ClearLink href={content[1].url} className={`baseline`}>
            <span
                data-tooltip
                style={{cursor: `unset`, fontWeight: `bold`}}
                className={`baseline tiny button-rd`} title={`Baseline experiments`}>Baseline</span>
          {content[1].text}
        </ClearLink>
        <ClearLink
            href={content[2].url}
            className={`differential padding-left-medium`}>
              <span
                  data-tooltip
                  style={{cursor: `unset`, fontWeight: `bold`}}
                  className={`differential tiny button-rd`}
                  title={`Differential experiments`}>Differential</span>
          {content[2].text}
        </ClearLink>
      </p>
    }
  </div>

SpeciesCard.propTypes = cardPropTypes

export default SpeciesCard
