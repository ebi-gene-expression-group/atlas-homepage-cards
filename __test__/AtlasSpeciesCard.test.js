import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import EbiSpeciesIcon from 'react-ebi-species'

import { batmanFilmsSpeciesCardProps, findingNemoSpeciesCardProps } from './TestUtils'
import AtlasSpeciesCard from '../src/cards/AtlasSpeciesCard'

Enzyme.configure({ adapter: new Adapter() })

describe(`AtlasSpeciesCard`, () => {
  test.each([ [batmanFilmsSpeciesCardProps.description.text, batmanFilmsSpeciesCardProps],
              [findingNemoSpeciesCardProps.description.text, findingNemoSpeciesCardProps] ])(
    `matches snapshot: %s`, (titleText, props) => {
    const tree = renderer.create(<AtlasSpeciesCard {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`does not render optional empty title and content`, () => {
    const wrapper = shallow(<AtlasSpeciesCard iconType={`species`} iconSrc={``} />)

    expect(wrapper.find(EbiSpeciesIcon).exists()).toBe(true)
    expect(wrapper.find(`h5`).exists()).toBe(false)
    expect(wrapper.find(`li`).exists()).toBe(false)
    expect(wrapper.find(`ul`).exists()).toBe(false)
  })

  test(`renders different experiment types with url`, () => {
    const props = batmanFilmsSpeciesCardProps
    props.description.url = `#`

    const wrapper = shallow(<AtlasSpeciesCard {...props}/>)

    expect(wrapper.find(`.baseline`)).toHaveLength(2)
    expect(wrapper.find(`.differential`)).toHaveLength(2)
  })
})
