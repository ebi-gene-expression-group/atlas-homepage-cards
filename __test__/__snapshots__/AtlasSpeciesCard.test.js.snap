// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtlasSpeciesCard matches snapshot: Batman films 1`] = `
<div
  style={
    Object {
      "marginBottom": 0,
      "paddingBottom": "2rem",
      "textAlign": "center",
    }
  }
>
  <span
    style={
      Object {
        "fontSize": "6rem",
      }
    }
  >
    <span
      className="icon icon-species"
      data-icon="("
      style={
        Object {
          "color": "indianred",
        }
      }
      title="Bat"
    />
  </span>
  <h5>
    Batman films
  </h5>
  <p
    className="experiments"
  >
    Batman (1943)
     
    <br />
    <a
      className="baseline sc-htpNat dbsHLq"
      href="https://en.wikipedia.org/wiki/Batman_and_Robin_(serial)"
    >
      <span
        className="baseline tiny button-rd"
        data-tooltip={true}
        style={
          Object {
            "cursor": "unset",
            "fontWeight": "bold",
          }
        }
        title="Baseline experiments"
      >
        Baseline
      </span>
      Batman and Robin (1949)
    </a>
    <a
      className="differential padding-left-medium sc-htpNat dbsHLq"
      href="https://en.wikipedia.org/wiki/Batman_(1966_film)"
    >
      <span
        className="differential tiny button-rd"
        data-tooltip={true}
        style={
          Object {
            "cursor": "unset",
            "fontWeight": "bold",
          }
        }
        title="Differential experiments"
      >
        Differential
      </span>
      Batman (1966)
    </a>
  </p>
</div>
`;

exports[`AtlasSpeciesCard matches snapshot: Finding Nemo 1`] = `
<div
  style={
    Object {
      "marginBottom": 0,
      "paddingBottom": "2rem",
      "textAlign": "center",
    }
  }
>
  <a
    href="https://en.wikipedia.org/wiki/Finding_Nemo"
    style={
      Object {
        "borderBottom": 0,
        "fontSize": "6rem",
      }
    }
  >
    <span
      className="icon icon-species"
      data-icon="Z"
      style={
        Object {
          "color": "deepskyblue",
        }
      }
      title="Zebrafish"
    />
  </a>
  <h5>
    <a
      href="https://en.wikipedia.org/wiki/Finding_Nemo"
    >
      Finding Nemo
    </a>
  </h5>
  <p
    className="experiments"
  >
    Dory
     
    <br />
    <a
      className="baseline sc-htpNat dbsHLq"
    >
      <span
        className="baseline tiny button-rd"
        data-tooltip={true}
        style={
          Object {
            "cursor": "unset",
            "fontWeight": "bold",
          }
        }
        title="Baseline experiments"
      >
        Baseline
      </span>
      Nemo
    </a>
    <a
      className="differential padding-left-medium sc-htpNat dbsHLq"
    >
      <span
        className="differential tiny button-rd"
        data-tooltip={true}
        style={
          Object {
            "cursor": "unset",
            "fontWeight": "bold",
          }
        }
        title="Differential experiments"
      >
        Differential
      </span>
      Gill
    </a>
  </p>
</div>
`;
